src/software/ai/navigator/navigator_private.cpp

```diff
@@ -0,0 +1,7 @@
+#include "software/ai/navigator/navigator_private.h"
+
+double calculateTransitionSpeedBetweenSegments(const Point &p1, const Point &p2,
+                                               const Point &p3, double final_speed)
+{
+    return final_speed * (p2 - p1).normalize().project((p3 - p2).normalize()).length();
```

Comment:
I have the same concerns about this as before. 

This method doesn't consider robot dynamics and breaks down for segments that have negative projections.

So long as we know the limitations, everything is good with me. I just wanted to point it out again.

Reply:
@garethellis0

Reply:
Yeah, totally agree. Re-doing the navigator to be spline based with explicit considerations for robot dynamics is definitely on the TODO list for this year, likely as part of the control changes, hence why I'm OK with leaving it for now.

Reply:
Sounds good to me. But there is also no chance that the navigator gives us a negative projection here?

Reply:
No I don't believe so. If you can think of a case, I'm happy to add a test for it

Reply:
what do you mean negative projection @EvanMorcom? Just negative values or something else?

Reply:
@MathewMacDougall 
If the path 'doubles' back on itself even slightly you will get a negative value for the projection. This means that the intermediate velocity is in the opposite direction you are traveling.

Sounds like that would be an issue to me.

Reply:
You won't ultimately get a negative value because even though you get a vector pointing in the "wrong direction", we ultimately take the length of it which will always return a positive value. I agree that this could still result in incorrect calculations since in cases you would want to go super slow (to totally change direction), but you at least won't get a literal negative number.

@jonathanlew can we make a ticket to revisit this so we don't forget later depending how stuff goes with splines?

Reply:
ticket: #1052